src/charm.py

```diff
@@ -346,36 +339,48 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         self._holistic_handler(event)
         self._on_oauth_integration_created(event)
 
-    def _on_public_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+    def _on_internal_ingress_joined(self, event: RelationJoinedEvent) -> None:
         self.unit.status = MaintenanceStatus("Configuring resources")
-        self._holistic_handler(event)
-        self._on_oauth_integration_created(event)
-        self._on_hydra_endpoints_ready(event)
+        self.internal_ingress._relation = event.relation
```

Comment:
If I understand correctly, the only reason we need this is to cater to unit tests and (MAYBE) edge cases around deferred events. While not ideal, I am fine with keeping this as we don't know how long it would take the mesh team to fix this.

BUT let's add a comment clarifying this for future reference and I think you should try adding this to the other ingress handlers as well to avoid adding hacks to the tests (like `reset_mocks` and mocking `secured`)